### PR TITLE
Create Docker Environment

### PR DESCRIPTION
**Pull request by christophermbooth**
This PR creates a docker-compose.yml in the root directory and a multi-stage Dockerfile in the villainary-react directory.  This may simplify developer onboarding as they should only need Docker installed on their machines in order to start contributing. A new readme containing directions for starting the app has been added to the root directory.  Hot reload is supported and changes are reflected in browser as they are made in source code.

Some components were blockers for the build process as they were not set to be client side components.  As we do not currently have a backend, I have converted the components to client-side components and disabled SSR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Villainary
+Villainary is a technical project built with Next.js, aiming to showcase an array of intriguing villains.  The project may grow to include various backend and frontend technologies to demonstrate mastery across multiple paradigms.
```
Comment (dweill) [inline README.md:2]: This is not accurate if being added at the root level. Only villainary react is developed using next JS. I see the second line includes information about multiple paradigms, but I think the overall description misses the point of villainary. Please relabel and rescope this to villainary-react

File: villainary-react/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# Stage 2: Build the Next.js application
+FROM node:18-alpine AS builder
+WORKDIR /app/villainary-react
+
+COPY --from=deps /app/node_modules ./node_modules
+COPY . .
+
+WORKDIR /app/villainary-react
+
+ENV NEXT_TELEMETRY_DISABLED 1
```
Comment (dweill) [inline villainary-react/Dockerfile:19]: could you provide a comment for lines like this. Not clear what this is doing at all.

File: villainary-react/app/profile/page.tsx [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import { Container } from '@mui/material';
-import Profile from './Profile';
+import dynamic from "next/dynamic";
```
Comment (dweill) [inline villainary-react/app/profile/page.tsx:2]: what does this do?

**Comment by dweill** [comment]
Can you provide a link or some additional context for why to build with docker we would need to disable SSR. I'm not opposed I just want to understand why it's necessary.

File: villainary-react/app/profile/page.tsx (hunk 3 above)
Comment (christophermbooth) [reply villainary-react/app/profile/page.tsx:2]: This is how next.js handles lazy loading of components.  It's also allowed me to explicitly label components as client side components.  Read more here: https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading

File: villainary-react/Dockerfile (hunk 2 above)
Comment (christophermbooth) [reply villainary-react/Dockerfile:19]: Comment added.

File: villainary-react/app/profile/page.tsx (hunk 3 above)
Comment (dweill) [reply villainary-react/app/profile/page.tsx:2]: is this different than "use client"?

**Review by dweill — APPROVED** [review]

**Pull request merged**